src/core/net/ip4_address.hpp

```diff
@@ -36,20 +36,36 @@
 
 #include "openthread-core-config.h"
 
+#include <openthread/nat64.h>
+
+#include "common/as_core_type.hpp"
 #include "common/clearable.hpp"
+#include "common/encoding.hpp"
 #include "common/equatable.hpp"
 #include "common/error.hpp"
 #include "common/string.hpp"
 
 namespace ot {
+
+namespace Ip6 {
+// Forward declaration for SynthesizeFromIp4Address
+class Address;
+} // namespace Ip6
```

Comment:
We can remove `// namespace Ip6` (since it is less than 10 lines away).
```suggestion
}
```